server/handler/export.go

```diff
@@ -0,0 +1,104 @@
+package handler
+
+import (
+	"database/sql"
+	"encoding/csv"
+	"net/http"
+	"strconv"
+
+	"github.com/go-sql-driver/mysql"
+	"github.com/src-d/gitbase-playground/server/serializer"
+	"github.com/src-d/gitbase-playground/server/service"
+)
+
+// Export returns a function that forwards an SQL query to gitbase and returns
+// the rows as CSV file
+func Export(db service.SQLDB) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		err := func(w http.ResponseWriter, r *http.Request) error {
+			query := r.URL.Query().Get("query")
+			if query == "" {
+				return serializer.NewHTTPError(http.StatusBadRequest,
+					`Bad Request. Expected body: { "query": "SQL statement" }`)
+			}
+
+			rows, err := db.Query(query)
```

Comment:
`handler.Query()` and `handler.Export()` methods are almost the same... could not we abstract the common parts?

Reply:
they are not that much the same. Most of the code of both handlers is switch block. But for `query` we return the original type (int, bool, ... in json) and for `export` we have to transform everything to string.
It's possible to "reuse" code. But then it would convert sql result to correct types -> correct types to string. 2 huge switch blocks instead of 1.

Reply:
fair enough

Reply:
fair enough

Reply:
though we might create service for querying DB later. Then we can move this logic there.